tools/wptrunner/wptrunner/testdriver-extra.js

```diff
@@ -16,27 +16,52 @@
         }
 
         if (data.status === "success") {
-            result = JSON.parse(data.message).result
+            result = JSON.parse(data.message).result;
             pending_resolve(result);
         } else {
             pending_reject(`${data.status}: ${data.message}`);
         }
     });
 
-    const get_frame = function(element, frame) {
-        let foundFrame = frame;
-        let frameDocument = frame == window ? window.document : frame.contentDocument;
-        if (!frameDocument.contains(element)) {
-          foundFrame = null;
-          let frames = document.getElementsByTagName("iframe");
-          for (let i = 0; i < frames.length; i++) {
-            if (get_frame(element, frames[i])) {
-              foundFrame = frames[i];
-              break;
+    let last_window_id = 0;
+    function get_window_id(win) {
+        if (win === window) {
+            return null;
+        }
+        // This is a hack until some implementations support proper window ids
+        // It won't work cross-frame
+        if (!win.name) {
+            win.name = "__wptrunner_window " + last_window_id++;
+        }
+        return win.name;
+    }
+
+    const get_context = function(element) {
+        console.log("get_context", element);
+        if (!element) {
+            return null;
+        }
+        let elementWindow = element.ownerDocument.defaultView;
+        let top = elementWindow.top;
+        if (top === elementWindow && elementWindow === window) {
+            // For the current window just return null
+            return null;
+        }
+        let stack = [[{window: top, index:-1}]];
+        while (stack.length) {
+            let item = stack.pop();
+            let target = item[item.length - 1];
```

Comment:
I'd call this `currentWindow` or `currentNode` (if we lean into describing it as a tree-walk), as `target` to me is actually `elementWindow` (as that's what we're trying to find).